.github/actions/pre_make/action.yml

```diff
@@ -0,0 +1,44 @@
+name: "Pre-Make"
+runs:
+  using: composite
+  steps:
+    - name: Restore make from cache
+      id: cache-make
+      uses: actions/cache@v3
+      with:
+        path: make.bin
+        key: v13_make.bin-${{ hashFiles('**/make.md5') }}
+    - name: Install make
+      if: steps.cache-make.outputs.cache-hit != 'true'
+      run: |
+        if [ -s make.bin ] ; then
+          echo "make.bin exists; not installing"
+        else
+          echo "/usr/bin/make doesn't exist; installing"
+          sudo apt-get update
+          sudo apt-get install -y make
+          sudo chmod -R 777 /usr/bin/make
+          sudo cp -f /usr/bin/make make.bin
+        fi
+        sudo cp -f make.bin /usr/bin/make
```

Comment:
This line here is a bit concerning to me, because if `actions/cache@v3` works like I expect it to, I'd expect this to never run on a cache hit, which makes me wonder what's actually installing `make` (and `dot`) in the case that they do get cached?

Would this just imply that on cache hits, we're not actually installing either?